src/EventListener/StorageListener.php

```diff
@@ -99,7 +99,11 @@ public function preHydrate(HydrationEvent $event)
 				foreach ($localeData as $key => $value) {
 					if (isset($contentType['fields'][$key]['type']) && $contentType['fields'][$key]['type'] !== 'repeater') {
 						$subject[$key] = is_array($value) ? json_encode($value) : $value;
-					}
+					}  elseif ( $key == 'templatefields') {
+						foreach ($value as $i => $field) {
```

Comment:
`$i` is the field name here, right? a more descriptive naming would be nice, `$i` is usually just used for indexes on arrays